app/qna/llm.py

```diff
@@ -98,12 +106,24 @@ def make_qna_chain():
 
     chain_type_kwargs = {"prompt": prompt}
 
-    chain = RetrievalQA.from_chain_type(
-        llm=OpenAI(),
-        chain_type="stuff",
-        retriever=redis.as_retriever(),
-        return_source_documents=True,
-        chain_type_kwargs=chain_type_kwargs
-    )
+    if OPENAI_API_TYPE=="azure":
```

Comment:
Can we pull this into a function like this:

`get_llm() -> LLM:`

and return the appropriate LLM based on the env vars?

Then just pass the llm object into the RetrievalQA chain

Reply:
done